7_factory_wrap/node-addon-api/addon.cc

```diff
@@ -0,0 +1,13 @@
+#include <napi.h>
+#include "myobject.h"
+
+Napi::Object CreateObject(const Napi::CallbackInfo& info) {
+  return MyObject::NewInstance(info.Env(), info[0]);
+}
+
+Napi::Object InitAll(Napi::Env env, Napi::Object exports) {
+  MyObject::Init(env, exports);
+  return Napi::Function::New(env, CreateObject, "CreateObject");
```

Comment:
Since exports is used within MyObject.Init() I don't think this is correct.  I think it should be something like

Napi:Object new_exports = Napi::Function::New(env, CreateObject, "CreateObject");
return MyObject:Init(env, exports)